engine/api/workflow_queue_staticfiles.go

```diff
@@ -0,0 +1,197 @@
+)
+
+func (api *API) getStaticFilesStoreHandler() service.Handler {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+		//TODO: to delete when swift will be available with auto-extract and temporary url middlewares
+		store := objectstore.Instance()
+		store.TemporaryURLSupported = false
+		return service.WriteJSON(w, store, http.StatusOK)
+	}
+}
+
+func (api *API) postWorkflowJobStaticFilesHandler() service.Handler {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+		db := api.mustDB()
+		// Load  Existing workflow Run Job
+		nodeJobRunID, errI := requestVarInt(r, "permID")
+		if errI != nil {
+			return sdk.WrapError(sdk.ErrInvalidID, "Invalid node job run ID")
+		}
+
+		vars := mux.Vars(r)
+		name := vars["name"]
+
+		_, params, errM := mime.ParseMediaType(r.Header.Get("Content-Disposition"))
+		if errM != nil {
+			return sdk.WrapError(errM, "Cannot read Content Disposition header")
+		}
+		fileName := params["filename"]
+
+		//parse the multipart form in the request
+		if err := r.ParseMultipartForm(100000); err != nil {
+			return sdk.WrapError(err, "Error parsing multipart form")
+		}
+		//get a ref to the parsed multipart form
+		m := r.MultipartForm
+
+		var entrypoint string
+		if len(m.Value["entrypoint"]) > 0 {
+			entrypoint = m.Value["entrypoint"][0]
+		}
+
+		if fileName == "" {
+			log.Warning("postWorkflowJobStaticFilesHandler> %s header is not set", "Content-Disposition")
+			return sdk.WrapError(sdk.ErrWrongRequest, "%s header is not set", "Content-Disposition")
+		}
+
+		nodeJobRun, errJ := workflow.LoadNodeJobRun(db, api.Cache, nodeJobRunID)
+		if errJ != nil {
+			return sdk.WrapError(errJ, "Cannot load node job run")
+		}
+
+		staticFile := sdk.StaticFiles{
+			Name:         name,
+			EntryPoint:   entrypoint,
+			NodeRunID:    nodeJobRun.WorkflowNodeRunID,
+			NodeJobRunID: nodeJobRunID,
+		}
+
+		files := m.File[fileName]
+		if len(files) == 1 {
+			file, err := files[0].Open()
+			if err != nil {
+				return sdk.WrapError(err, "cannot open file")
+			}
+			defer file.Close()
+
+			publicURL, err := objectstore.ServeStaticFiles(&staticFile, staticFile.EntryPoint, file)
+			if err != nil {
+				return sdk.WrapError(err, "Cannot serve static files in store")
+			}
+			staticFile.PublicURL = publicURL
+		}
+
+		if err := workflow.InsertStaticFiles(db, &staticFile); err != nil {
+			_ = objectstore.Delete(&staticFile)
+			return sdk.WrapError(err, "Cannot insert static files in database")
+		}
+		return service.WriteJSON(w, staticFile, http.StatusOK)
+	}
+}
+
+func (api *API) postWorkflowJobStaticFilesWithTempURLHandler() service.Handler {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+		if !objectstore.Instance().TemporaryURLSupported {
+			return sdk.ErrForbidden
+		}
+
+		store, ok := objectstore.Storage().(objectstore.DriverWithRedirect)
+		if !ok {
+			return sdk.WrapError(sdk.ErrForbidden, "cast error")
```

Comment:
FYI: If you want to add more info on a error, that will be printed in log and error message returned by the api, you can use sdk.NewErrorFrom(sdk.ErrForbidden, "A cast error occured")

Reply:
done